### ISSUE TITLE
SIP Via Header on 100 Trying

### ISSUE THREAD

**Issue by wpichler** [signalwire/freeswitch#319]
Hi all,

following behaviour on freeswitch  1.10.1 - when getting an inbound call from a system behind a sip proxy - then the invite does have 2 Via Headers (Top Via Header from Proxy, second Via Header from system behind proxy). Freeswitch does send a 100 Trying - with only the Top Via Header set. This does cause the proxy to not forward the via header - so we get a second invite and so on.

If freeswitch does answer the call - then it will send a 200 Ok with all Via Headers.

Why does it not send the Via headers in 100 Trying ? Is there a config option for it (i did not found one) ?

In the RFC3261 - 8.2.6.2 is the following:
"The Via header
   field values in the response MUST equal the Via header field values
   in the request and MUST maintain the same ordering."

So i think freeswitch (libsofia) is acting wrong here...

**Comment by briankwest** [comment]
Please attach a complete sip trace with debug log.

**Comment by lazedo** [comment]
maybe this is relevant [100 responses are not forwarded by proxies, so only include the topmost Via header](https://github.com/signalwire/freeswitch/blob/master/libs/sofia-sip/libsofia-sip-ua/nta/nta.c#L6465) from this https://github.com/signalwire/freeswitch/commit/6ed86abf9f6ab1116a39b28c651dec1d4f6cf89f

**Comment by wpichler** [comment]
@lazedo Yep - this could be the cause for the problem - i will test tomorrow by disabling this part:

    if (sip->sip_status && sip->sip_status->st_status == 100)
      sip->sip_via = (sip_via_t *)msg_header_copy_one(home, (msg_header_t const *)irq->irq_via);
    else
      sip->sip_via = sip_via_copy(home, irq->irq_via);

only using 
      sip->sip_via = sip_via_copy(home, irq->irq_via);

@armenfirespotter
This part is the nta_incoming_reponse_headers function - so this is in a response to a proxy which did already forwarded the Via Headers - or which inserted the via headers. Why should it here be necessary to remove a Via Header ?

**Comment by armenfirespotter** [comment]
@lazedo @wpichler this was awhile ago, but the point of FS-9300 was to introduce the acknowledge_call app. At the time, IIRC freeswitch/sofia did not follow the RFC spec re the quoted bit above: it only copied one Via header in a 100 Trying, not the whole Via stack as RFC-compliant implementation should. w.r.t this implementation, I wanted to preserve existing on-the-wire behavior of freeswitch/sofia for compatibility with existing (possibly-non-rfc-compliant) implementations. In retrospect, my comment kinda sucks and doesn't tell this story.

FWIW, I have no objection with bringing this implementation in-line with RFC compliance, perhaps by deleting lines 6465 to 6468, inclusive.

**Comment by wpichler** [comment]
I've disabled this part - but still the problem exists. So there must be an other point where the sip via headers gets set. In this function it does only work on the via header if it is not already set.

In which part of the code does the SIP Via header for a 100 Trying response are getting set ?

**Comment by grantbagdasarian** [comment]
Any progress on this matter? Or perhaps a workaround?

**Comment by eduardofoliveira** [comment]
Is it possible to enable some flag in the profile to keep the 2 headers via in Trying?

**Comment by kriegschrei** [comment]
Is there any update on this ticket?  The stripping of the Via headers impacts the ability to integrate with stateless proxies that rely on the Via headers.

**Comment by gregoriusus** [comment]
We have same issue. Proxy keep sending invite

**Comment by eduardofoliveira** [comment]
Hi,

Is there any way to get around this problem?

I can't use recent versions of Freeswitch because calls are dropped if there is no ringback within 4s

![Image](https://github.com/user-attachments/assets/fcd2d7f0-0564-4bf5-84ff-86f1ca4b9d8c)

I'm using version: FreeSWITCH Version 1.6.20

Until there is a solution in more updated versions

**Comment by heywood-y** [comment]
We have same issue.

**Comment by fwittenberg** [comment]
Hello everyone,

I’ve noticed that the patch for the Via header issue with 100 Trying responses (see freeswitch/sofia-sip#41) is already available as a code diff. This is directly relevant for us, as it’s causing routing problems in our contact center.

Could you let us know if and when this fix will be merged into the official FreeSWITCH releases?
Is there a planned merge date or a roadmap we can refer to?

Thank you very much in advance for your response!
